avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroMapper.java

```diff
@@ -28,12 +47,13 @@ public AvroMapper() {
     public AvroMapper(AvroFactory f) {
         super(f);
         registerModule(new AvroModule());
+        setDefaultTyping(new AvroSchemaTypeResolverBuilder());
```

Comment:
Avro puts all of its type information in the schema and none in the data structure itself, so registering as the default typer seemed to be the easiest way to make Jackson check the schema for type information since there won't be any annotations to indicate type information in situations like deserializing into a `Map<String,Object>`.